cmd/dep/root_analyzer.go

```diff
@@ -53,6 +57,71 @@ func (a *rootAnalyzer) InitializeRootManifestAndLock(dir string, pr gps.ProjectR
 	return
 }
 
+func (a *rootAnalyzer) cacheDeps(pr gps.ProjectRoot, pkgT pkgtree.PackageTree) {
+	packages := make(map[string]bool)
+	dependencies := make(map[gps.ProjectRoot][]string)
+	logger := a.ctx.Err
+
+	syncDep := func(pr gps.ProjectRoot, sm gps.SourceManager) {
+		message := fmt.Sprintf("Cached %s", pr)
+		if err := sm.SyncSourceFor(gps.ProjectIdentifier{ProjectRoot: pr}); err != nil {
+			message = fmt.Sprintf("Unable to cache %s - %s", pr, err)
+		}
+		logger.Printf(message)
+	}
+
+	for _, v := range pkgT.Packages {
+		// TODO: Some errors maybe should not be skipped ;-)
+		if v.Err != nil {
+			logger.Printf("%v", v.Err)
+			continue
+		}
+
+		logger.Printf("Package %q, analyzing...", v.P.ImportPath)
+
+		for _, ip := range v.P.Imports {
+			if isStdLib(ip) {
+				continue
+			}
+			if hasImportPathPrefix(ip, string(pr)) {
+				continue
+			}
+
+			pr, err := a.sm.DeduceProjectRoot(ip)
+			if err != nil {
+				// return nil, nil, nil
+			}
+
+			packages[ip] = true
+			if _, ok := dependencies[pr]; ok {
+				if !contains(dependencies[pr], ip) {
+					dependencies[pr] = append(dependencies[pr], ip)
+				}
+				continue
+			}
+
+			go syncDep(pr, a.sm)
+			dependencies[pr] = []string{ip}
+		}
+	}
+}
+
+func hasImportPathPrefix(s, prefix string) bool {
+	if s == prefix {
+		return true
+	}
+	return strings.HasPrefix(s, prefix+"/")
+}
+
+func isStdLib(path string) bool {
```

Comment:
This function is already defined in `github.com/golang/dep/internal/gps/paths` as `IsStandardImportPath`.